src/App.js

```diff
@@ -1,19 +1,20 @@
-import React from 'react';
+import React, {useState} from 'react';
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
 
 function App() {
+  const [todoList, setTodoList] = useState([]);
 
-  const [newTodo, setNewTodo] = React.useState('');
-
-    return (
-      <div>
-        <h1 >To Do List</h1>
-        <AddTodoForm onAddTodo ={setNewTodo}/>
-        <p>New Todo: {newTodo}</p>
-        <TodoList />      
-      </div>
+  function addTodo (newTodo) {
+    setTodoList ([...todoList, newTodo])
```

Comment:
Nitpick (doesn’t require  changes):
While this code works as intended, there's a bug in how its implemented. To be brief, React batches (groups) state updates. Therefore, if you have something like this:
```js
// const [count, setCount] = useState(0);
setCount(count + 1) // expected 0 + 1
setCount(count + 1) // expected 1 + 1
setCount(count + 1) // expected 2 + 1
// expected count = 3
```
You may think the count is being set to 3, but it's not! Instead, since React batches those state setters, it takes the current count value (0) and increases it by 1 multiple times:
```js
// const [count, setCount] = useState(0);
setCount(count + 1) // actual 0 + 1
setCount(count + 1) // actual 0 + 1
setCount(count + 1) // actual 0 + 1
// actual count = 1
```

To mitigate this bug, you'll want to use a **function** inside of the state setter whose first argument is the previous state (`prevCount`) and returns the next state (`prevCount + 1`):
```js
// const [count, setCount] = useState(0);
setCount(prevCount => prevCount + 1) // actual 0 + 1
setCount(prevCount => prevCount + 1) // actual 1 + 1
setCount(prevCount => prevCount + 1) // actual 2 + 1
// actual count = 3
```
---
With the above knowledge, how might you change this code below to reflect the above?
```js
setTodoList([...todoList,newTodo])
```

<details>
<summary>Click here for answer if you're stuck</summary>

```js
setTodoList(prevTodoList => ([ ...prevTodoList, newTodo ]))
```
</details>

Reply:
Thank you Matt, I didn't realize the difference at once. It's good to know.